src/middleware/validator.js

```diff
@@ -1,105 +1,31 @@
 
-const createValidationSchema = validationMap => {
-  if (!validationMap) {
-    throw new Error('No validation rules supplied')
-  }
-
-  const validations = validationMap
-  const schemaObject = {}
-
-  if (validations) {
-    Object.keys(validations).forEach(key => {
-      let rule = validations[`${key}`]
-      switch (rule.type) {
-        case 'string':
-          if (rule.required) {
-            schemaObject[`${key}`] = Joi.string().required()
-          } else if (rule.default) {
-            schemaObject[`${key}`] = Joi.string().default(rule.default)
-          } else {
-            schemaObject[`${key}`] = Joi.string()
-          }
-          break
-
-        case 'number':
-          if (rule.required) {
-            schemaObject[`${key}`] = Joi.number().required()
-          } else if (rule.default) {
-            schemaObject[`${key}`] = Joi.number().default(rule.default)
-          } else {
-            schemaObject[`${key}`] = Joi.number()
-          }
-          break
-
-        case 'boolean':
-          if (rule.required) {
-            schemaObject[`${key}`] = Joi.boolean().required()
-          } else if (rule.default) {
-            schemaObject[`${key}`] = Joi.boolean().default(rule.default)
-          } else {
-            schemaObject[`${key}`] = Joi.boolean()
-          }
-          break
+const configurations = config.getConfig()
 
-        case 'array':
-          if (rule.required) {
-            schemaObject[`${key}`] = Joi.array().required()
-          } else if (rule.default) {
-            schemaObject[`${key}`] = Joi.array().default(rule.default)
-          } else {
-            schemaObject[`${key}`] = Joi.array()
-          }
-          break
-
-        case 'object':
-          if (rule.required) {
-            schemaObject[`${key}`] = Joi.object().required()
-          } else if (rule.default) {
-            schemaObject[`${key}`] = Joi.object().default(rule.default)
-          } else {
-            schemaObject[`${key}`] = Joi.object()
-          }
-          break
-
-        default:
-          throw new Error(`Validation rule type is not supported: ${rule.type}`)
-      }
-    })
-
-    if (schemaObject) {
-      return Joi.object(schemaObject)
-    }
-
-    throw new Error('Joi validation schema creation failed')
+const performValidation = (ctx, schema) => {
+  if (!schema) {
+    throw new Error(`No validation rules supplied`)
   }
 
-  throw new Error('No validation schema in file')
-}
+  if (!ctx || !ctx.request || !ctx.request.body) {
+    throw new Error(`Invalid request body`)
+  }
 
-const performValidation = (ctx, schema) => {
-  const {error, value} = schema.validate(ctx.request.body)
+  const ajv = new Ajv({
+    nullable: configurations.acceptNullValues
+  })
+  const valid = ajv.validate(schema, ctx.request.body)
 
-  if (error) {
-    throw new Error(`Validation execution failed: ${error.message}`)
+  if (!valid) {
+    throw new Error(`Validation execution failed: ${ajv.errorsText()}`)
```

Comment:
@bradsawadye I don't think the word execution shoudl be here since the execution of the validation worked. The actual request did not pass validation.

```suggestion
    throw new Error(`Validation failed: ${ajv.errorsText()}`)
```

Reply:
addressed in https://github.com/jembi/openhim-mediator-mapping/pull/21/commits/4a37f5f33a74753cc5a9f11705407cb9b974570e